### PR TITLE
Adding some refactoring tests with tuples

### PR DESCRIPTION
**Pull request by jcouv**
Closes https://github.com/dotnet/roslyn/issues/10585 and https://github.com/dotnet/roslyn/issues/10583

@CyrusNajmabadi for review (post-dev15).

### CODE REVIEW

**Comment by alrz** [comment]
I'd suggest a test where the abstract class/interface to implement is not in the same source, e.g. #16434

**Comment by jcouv** [comment]
@alrz The test that you suggested seems to pass. If you don't mind, take a look to confirm this is what you meant. Thanks https://github.com/dotnet/roslyn/pull/16663/commits/e64f4a92828d6842d1aba8dc934d9cb8a6444383

**Comment by alrz** [comment]
@jcouv In my my scenario the two types were not in the same assembly, however, you've said in #12949 

>I have a fix and test for this scenario with a tuple with names in a method coming from metadata.

So I supposed it's just dev14 and it's no longer an issue.

**Comment by jcouv** [comment]
Cool. Thanks

**Comment by jcouv** [comment]
@CyrusNajmabadi @dotnet/roslyn-ide for review (test-only change).

File: src/EditorFeatures/CSharpTest/Diagnostics/GenerateType/GenerateTypeTests.cs [hunk 1]
```diff
@@ -268,6 +268,31 @@ public Foo()
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateType)]
+        public async Task TestGenerateClassWithCtorFromObjectCreationWithTuple()
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Diagnostics/GenerateType/GenerateTypeTests.cs]: Can we have tests with named tuples as well?

Comment (jcouv) [inline src/EditorFeatures/CSharpTest/Diagnostics/GenerateType/GenerateTypeTests.cs]: Added

File: src/EditorFeatures/VisualBasicTest/ImplementInterface/ImplementInterfaceCommandHandlerTests.vb [hunk 2]
```diff
@@ -70,6 +70,36 @@ End Interface")
...
+Imports System
+
+Class Foo
+    Implements IFoo$$
+End Class
+Interface IFoo
+    Function Method(x As (Alice As Integer, Bob As Integer)) As (String, String)
+End Interface</text>
+
+            Dim expectedText = <text>
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasicTest/ImplementInterface/ImplementInterfaceCommandHandlerTests.vb]: These should be in ImplementInterfaceTests.vb.  ImplementINterfaceCommandHandlerTests is for actually testing hte command handling infrastructure.

**Comment by CyrusNajmabadi** [comment]
👍 With just the tiny changes i mentioned.

File: src/EditorFeatures/VisualBasicTest/ImplementInterface/ImplementInterfaceTests.vb [hunk 3]
```diff
@@ -35,6 +35,36 @@ Class C
...
+        Throw New NotImplementedException()
+    End Function
+End Class
+Interface IFoo
+    Function Method(x As (Alice As Integer, Bob As Integer)) As (String, String)
+End Interface</text>
+
+            Await TestAsync(code,
+             expectedText,
+             Sub() Throw New Exception("The operation should have been handled."),
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasicTest/ImplementInterface/ImplementInterfaceTests.vb]: do you need these lambdas?

Comment (jcouv) [inline src/EditorFeatures/VisualBasicTest/ImplementInterface/ImplementInterfaceTests.vb]: Simplified. Thanks

**Comment by CyrusNajmabadi** [comment]
Great, thanks.  👍 on adding more tests :)

**Comment by jcouv** [comment]
Thanks for the review. Merging ahead.

**Pull request merged**